### PR TITLE
[WebNN EP] Create MLGraphBuilder for every model builder

### PR DESCRIPTION
**Pull request by Honry**
Currently WebNN spec only allows MLGraphBuilder.build() to be called once, we need to create new builder for every subgraph in WebNN EP.

Spec change: https://github.com/webmachinelearning/webnn/pull/717

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, @huningxin, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/model_builder.h [hunk 1]
```diff
@@ -22,9 +22,12 @@ class IOpBuilder;
...
  public:
   ModelBuilder(const GraphViewer& graph_viewer, const logging::Logger& logger,
-               const emscripten::val& context, const emscripten::val& builder,
-               const DataLayout preferred_layout, const WebnnDeviceType wnn_device_type);
-  ~ModelBuilder() = default;
+               const emscripten::val& context, const DataLayout preferred_layout,
+               const WebnnDeviceType wnn_device_type);
+  ~ModelBuilder() {
+    // Explicitly release the WebNN builder to free memory.
+    wnn_builder_ = emscripten::val::undefined();
```
Comment (huningxin) [inline onnxruntime/core/providers/webnn/builders/model_builder.h:29]: Given builder is not usable after graph built, should we release it right after calling `build()`?

**Review by huningxin — APPROVED** [review]
LGTM with a comment.

BTW, you may want to refer to the spec change PR: https://github.com/webmachinelearning/webnn/pull/717

File: onnxruntime/core/providers/webnn/builders/model_builder.h (hunk 1 above)
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/model_builder.h]: Thanks, good point, done!

**Review by huningxin — APPROVED** [review]
LGTM!

**Review by fdwr — APPROVED** [review]
👍

**Comment by fdwr** [comment]
@honry Merge with your most recent change?

**Comment by Honry** [comment]
> @Honry Merge with your most recent change for the Web CI failures?

Sure, done, pls. help retrigger the CI. Thanks!

**Pull request merged**